modules/core/04-channel/client/cli/tx.go

```diff
@@ -0,0 +1,160 @@
+package cli
+
+import (
+	"fmt"
+	"github.com/spf13/cobra"
+	"regexp"
+	"slices"
+	"strings"
+
+	"github.com/cosmos/cosmos-sdk/client"
+	"github.com/cosmos/cosmos-sdk/client/flags"
+	"github.com/cosmos/cosmos-sdk/client/tx"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/version"
+	govtypesv1 "github.com/cosmos/cosmos-sdk/x/gov/types/v1"
+
+	"github.com/cosmos/ibc-go/v8/modules/core/04-channel/types"
+	ibcexported "github.com/cosmos/ibc-go/v8/modules/core/exported"
+)
+
+const (
+	flagMetadata    = "metadata"
+	flagSummary     = "summary"
+	flagTitle       = "title"
+	flagJSON        = "json"
+	flagPortPattern = "port-pattern"
+	flagExpedited   = "expedited"
+	flagChannelIDs  = "channel-ids"
+)
+
+func newUpgradeChannelsCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "upgrade-channels",
+		Short: "Upgrade IBC channels",
+		Long: `Submit a governance proposal to upgrade all open channels whose port matches a specified pattern 
+(the default is transfer), optionally, specific an exact list of channel IDs with a comma separated list.`,
```

Comment:
```suggestion
(the default is transfer), optionally, an exact list of comma separated channel IDs may be specified.`,
```

Reply:
should we specify in this docstring that if channel IDs are specified then it must match the specified port? or leave that for the longer documentation?